api/src/pcapi/core/offerers/models.py

```diff
@@ -843,9 +844,17 @@ def legal_category(self) -> dict:
     def first_user(self) -> "users_models.User | None":
         # Currently there is no way to mark a UserOfferer as the owner/creator, so we consider that this first user
         # is the oldest entry in the table. When creator leaves the offerer, next registered user becomes the "first".
+        # The relation being preserved during a rejection or deletion, we always keep an active user
         if not self.UserOfferers:
             return None
-        return self.UserOfferers[0].user
+        not_inactive_user_offerers = [
+            u
+            for u in self.UserOfferers
+            if u.validationStatus != ValidationStatus.REJECTED and u.validationStatus != ValidationStatus.DELETED
+        ]
+        if not not_inactive_user_offerers:
+            return None
+        return not_inactive_user_offerers[0].user
```

Comment:
Suggestion : ne parcourir la liste que jusqu'au premier actif, puisqu'on n'a pas besoin de parcourir tous les utilisateurs liés, juste arriver au premier. L'impact en performance est faible mais mieux que rien. Et peut-être même un poil plus vite lisible :)
```suggestion
        for user_offerer in self.UserOfferers:
            if not (user_offerer.isRejected or user_offerer.isDeleted):
                return user_offerer.user
        return None
```